gallery/how_to/work_with_microtvm/micro_aot.py

```diff
@@ -0,0 +1,168 @@
+
+    tflite_model = tflite.Model.Model.GetRootAsModel(tflite_model_buf, 0)
+
+input_shape = (1, 49, 10, 1)
+INPUT_NAME = "input_1"
+relay_mod, params = relay.frontend.from_tflite(
+    tflite_model, shape_dict={INPUT_NAME: input_shape}, dtype_dict={INPUT_NAME: "int8"}
+)
+
+######################################################################
+# Defining the target
+# -------------------
+#
+# Now we need to define the target, runtime and executor. In this tutorial, we focused on
+# using AOT host driven executor. We use the host micro target which is for running a model
+# on X86 CPU using CRT runtime or running a model with Zephyr platform on qemu_x86 simulator
+# board. In the case of a physical microcontoller, we get the target model for the physical
+# board (E.g. nucleo_f746zg) and pass it to `tvm.target.target.micro` to create a full
+# micro target.
+#
+RUNTIME = Runtime("crt", {"system-lib": True})
+TARGET = tvm.target.target.micro("host")
+EXECUTOR = Executor("aot")
+
+if use_physical_hw:
+    boards_file = pathlib.Path(tvm.micro.get_microtvm_template_projects("zephyr")) / "boards.json"
+    with open(boards_file) as f:
+        boards = json.load(f)
+    BOARD = os.getenv("TVM_MICRO_BOARD", default="nucleo_f746zg")
+    TARGET = tvm.target.target.micro(boards[BOARD]["model"])
+
+######################################################################
+# Compile the model
+# -----------------
+#
+# Now, we compile the model for the target:
+#
+with tvm.transform.PassContext(opt_level=3, config={"tir.disable_vectorize": True}):
+    module = tvm.relay.build(
+        relay_mod, target=TARGET, params=params, runtime=RUNTIME, executor=EXECUTOR
+    )
+
+######################################################################
+# Create a microTVM project
+# -----------------------
+#
+# Now that we have the comipled model as an IRModule, we need to create a project
+# with the compiled model in microTVM. To do this, we use Project API. We have defined
+# CRT and Zephyr microTVM template projects which are used for X86 CPU and Zephyr platforms
+# respectively.
+#
+template_project_path = pathlib.Path(tvm.micro.get_microtvm_template_projects("crt"))
+project_options = {}  # You can use options to provide platform-specific options through TVM.
+
+if use_physical_hw:
+    template_project_path = pathlib.Path(tvm.micro.get_microtvm_template_projects("zephyr"))
+    project_options = {"project_type": "host_driven", "zephyr_board": BOARD}
+
+temp_dir = tvm.contrib.utils.tempdir()
+generated_project_dir = temp_dir / "project"
+project = tvm.micro.generate_project(
+    template_project_path, module, generated_project_dir, project_options
+)
+
+######################################################################
+# Build, flash and execute the model
+# -----------------------
+# Next, we build the microTVM project and flash it. Flash step is specific to
+# physical microcontrollers and it is skipped if it is using CRT runtime or running
```

Comment:
"and it is skipped if it is simulating a microcontroller via the host main.cc or if a Zephyr emulated board is selected as the target."